### PR TITLE
Adds `cond` method

### LINKED ISSUE #437: Consider adding `.cond` to `Result` container
[Either](https://typelevel.org/cats/datatypes/eithert.html) provides a way to avoid the boilerplate below:
```python
def func():
    if something:
        return Success(...)
    return Failure(...)
```

It provides the `.cond` method, with `Result` container will be something like this:
```python
def func():
    return Result.cond(
        my_condition,   # The boolean value
        success_value,  # If boolean is `True` the return will be Success(success_value)
        failure_value,  # If boolean is `False` the return will be Failure(failure_value)
    )
```

### PR DESCRIPTION
**Pull request by thepabloaguilar**
Working code example:
```python
>>> from returns.methods import cond
>>> from returns.result import Failure, Result, Success

>>> def is_numeric(string: str) -> Result[str, str]:
...     return cond(
...         Result,
...         string.isnumeric(),
...         'It is a number',
...         'It is not a number',
...     )

>>> assert is_numeric('42') == Success('It is a number')
>>> assert is_numeric('non numeric') == Failure('It is not a number')
```

Closes #437

### CODE REVIEW

File: typesafety/test_methods/test_cond.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+- case: cond_1
+  disable_cache: true
+  main: |
+    from returns.methods import cond
+    from returns.result import Result
+
+    reveal_type(cond(Result, True, 42, '42'))  # N: Revealed type is 'returns.result.Result*[builtins.int*, builtins.str*]'
```
Comment (sobolevn) [inline typesafety/test_methods/test_cond.yml:7]: Let's add `FutureResult`,`IOResult`, `ReaderIOResult`, `ReaderFutureResult`, and custom datatype test-cases.

**Comment by thepabloaguilar** [comment]
> 1. What do you think about point-free function for this method?

Can be handy, so we need to add `.cond(...)` method to our interface right??

```python
class ResultLikeN(...):
    @classmethod
    def cond(
        cls, boolean: bool, success_value: _ST, error_value: _ET,
    ) -> 'ResultLikeN[_ST, _ET, _TT]':
        if boolean:
            return cls.from_value(success_value)
        return cls.from_error(error_value)

def cond(
    boolean: bool,
    success_value: _ST,
    error_value: _ET,
) -> Callable[[_ResultKind], KindN[_ResultKind, _ST, _ET, _TT]]:
    return lambda container: container.cond(boolean, success_value, error_value)
```

> 2. Can you please add a changelog entry?

Yes, I can!

**Comment by sobolevn** [comment]
@thepabloaguilar no, I was asking about https://github.com/dry-python/returns/tree/master/returns/pointfree

**Comment by thepabloaguilar** [comment]
Oh, I didn't see that changes. I remember every pointfree function were implement around container methods!!
Now these functions are implemented around the functions from the `methods` package.

**Comment by thepabloaguilar** [comment]
So, the API should be:
```python
cond(Result, success, failure)(is_success)
```

**Comment by thepabloaguilar** [comment]
Does `point free cond` might be included in other PR?

**Comment by sobolevn** [comment]
@thepabloaguilar yes, please! 👍

File: typesafety/test_methods/test_cond.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    reveal_type(cond(ReaderIOResult, True, 1.0, False))  # N: Revealed type is 'returns.context.requires_context_ioresult.RequiresContextIOResult*[builtins.float*, builtins.bool*, <nothing>]'
+
+- case: cond_reader_future_result
+  disable_cache: true
+  main: |
+    from returns.methods import cond
+    from returns.context import ReaderFutureResult
+
+    reveal_type(cond(ReaderFutureResult, True, 1, 1.0))  # N: Revealed type is 'returns.context.requires_context_future_result.RequiresContextFutureResult*[builtins.int*, builtins.float*, <nothing>]'
```
Comment (sobolevn) [inline typesafety/test_methods/test_cond.yml:39]: Should be `returns.context.requires_context_future_result.RequiresContextFutureResult*[builtins.int*, builtins.float*, Any]`

File: returns/methods/cond.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+_ResultKind = TypeVar('_ResultKind', bound=ResultLikeN)
+
+
+def internal_cond(
+    container_type: Type[_ResultKind],
+    is_success: bool,
+    success_value: _ValueType,
+    error_value: _ErrorType,
+) -> KindN[_ResultKind, _ValueType, _ErrorType, _ThirdType]:
```
Comment (sobolevn) [inline returns/methods/cond.py:18]: Should be `Kind2`, because we don't work with the third type at all

Comment (sobolevn) [reply returns/methods/cond.py:18]: That's why it is always `never`

File: typesafety/test_methods/test_cond.yml (hunk 2 above)
Comment (thepabloaguilar) [reply typesafety/test_methods/test_cond.yml:39]: I think we have a `type` problem here or not!! It just work if I assign the result to a variable and typing it properly:
```yml
- case: cond_reader_ioresult
  disable_cache: true
  main: |
    from typing import Any

    from returns.methods import cond
    from returns.context import ReaderIOResult

    container: ReaderIOResult[float, bool, Any] = cond(ReaderIOResult, True, 1.0, False)
    reveal_type(container)  # N: Revealed type is 'returns.context.requires_context_ioresult.RequiresContextIOResult[builtins.float, builtins.bool, Any]'
```

Comment (thepabloaguilar) [reply typesafety/test_methods/test_cond.yml:39]: Sorry, it's working without the above solution!! It was `mypy` cache kidding me hahaha
I just deleted the cache an everything worked properly!!

**Pull request merged**